script.js

```diff
@@ -0,0 +1,159 @@
+    {code: 'Delete', specKey: true, name: 'Del' },
+
+    {code: 'CapsLock', specKey: true },  
+    {code: 'KeyA', en: 'a', enShift: 'A', ru: 'ф', ruShift: 'Ф'},
+    {code: 'KeyS', en: 's', enShift: 'S', ru: 'ы', ruShift: 'Ы'},
+    {code: 'KeyD', en: 'd', enShift: 'D', ru: 'в', ruShift: 'В'},
+    {code: 'KeyF', en: 'f', enShift: 'F', ru: 'а', ruShift: 'А'},
+    {code: 'KeyG', en: 'g', enShift: 'G', ru: 'п', ruShift: 'П'},
+    {code: 'KeyH', en: 'h', enShift: 'H', ru: 'р', ruShift: 'Р'},
+    {code: 'KeyJ', en: 'j', enShift: 'J', ru: 'о', ruShift: 'О'},
+    {code: 'KeyK', en: 'k', enShift: 'K', ru: 'л', ruShift: 'Л'},
+    {code: 'KeyL', en: 'l', enShift: 'L', ru: 'д', ruShift: 'Д'},
+    {code: 'Semicolon', en: ';', enShift: ':', ru: 'ж', ruShift: 'Ж'},
+    {code: 'Quote', en: "'", enShift: '"', ru: 'э', ruShift: 'Э'},
+    {code: 'Enter', specKey: true},
+
+    {code: 'ShiftLeft', specKey: true, name: 'Shift'},
+    {code: 'KeyZ', en: 'z', enShift: 'Z', ru: 'я', ruShift: 'Я'},
+    {code: 'KeyX', en: 'x', enShift: 'X', ru: 'ч', ruShift: 'Ч'},
+    {code: 'KeyC', en: 'c', enShift: 'C', ru: 'с', ruShift: 'С'},
+    {code: 'KeyV', en: 'v', enShift: 'V', ru: 'м', ruShift: 'М'},
+    {code: 'KeyB', en: 'b', enShift: 'B', ru: 'и', ruShift: 'И'},
+    {code: 'KeyN', en: 'n', enShift: 'N', ru: 'т', ruShift: 'Т'},
+    {code: 'KeyM', en: 'm', enShift: 'M', ru: 'ь', ruShift: 'Ь'},
+    {code: 'Comma', en: ',', enShift: '<', ru: 'б', ruShift: 'Б'},
+    {code: 'Period', en: '.', enShift: '>', ru: 'ю', ruShift: 'Ю'},
+    {code: 'Slash', en: '/', enShift: '?', ru: '.', ruShift: ','},
+    {code: 'ArrowUp', en: '↑', enShift: '↑', ru: '↑', ruShift: '↑'},
+    {code: 'ShiftRight', specKey: true, name: 'Shift'},
+
+    {code: 'ControlLeft', specKey: true, name: 'Ctrl'},
+    {code: 'MetaLeft', specKey: true, name: 'Win'},
+    {code: 'AltLeft', specKey: true, name: 'Alt'},
+    {code: 'Space', specKey: true, name: ''},
+    {code: 'AltRight', specKey: true, name: 'Alt'},
+    {code: 'ArrowLeft', en: '←', enShift: '←', ru: '←', ruShift: '←'},
+    {code: 'ArrowDown', en: '↓', enShift: '↓', ru: '↓', ruShift: '↓'},
+    {code: 'ArrowRight', en: '→', enShift: '→', ru: '→', ruShift: '→'},
+    {code: 'ControlRight', specKey: true, name: 'Ctrl'}
+  ];
+        
+  h1 = document.createElement('h1');
+  h1.innerHTML = 'Виртуальная клавиатура';
+  document.body.append(h1);
+
+  let textarea = document.createElement('textarea');
+  textarea.id = 'textarea';
+  textarea.autofocus = true;
+  document.body.append(textarea);
+
+  let keyboard = document.createElement('div');
+  keyboard.id = 'keyboard';
+  document.body.append(keyboard);
+
+  let addKeysDom = function (objKey) {
+    let key = document.createElement('div');
+    key.className = objKey.code;
+    key.innerText = objKey.en || objKey.name || objKey.code;
+    return document.getElementById('keyboard').append(key);
+  }
+
+  keys.forEach(objKey => addKeysDom(objKey));
+
+
+  class Button {
```

Comment:
This implementation of `class Button` is strict logic violation. It seems, you tried to implement class that returns array of buttons and add methods to proccess user events. This is the wrong way of using classes. In this case you had to create class that create one instance of button and then add that button instances to common array.

Example:
```
class Button {
  constructor(options) {
    /* Your implementation of each button */
  }
}

const buttons = keys.map(key => new Button(key));
```